service/history/handler.go

```diff
@@ -681,6 +697,10 @@ func (h *handlerImpl) StartWorkflowExecution(
 
 	startRequest := wrappedRequest.StartRequest
 	workflowID := startRequest.GetWorkflowID()
+
+	// TODO: actually rate limit
+	h.workflowIDCache.AllowExternal(domainID, workflowID)
```

Comment:
Is metric emission is somewhere inside? 
I expected seeing:
if  h.workflowIDCache.AllowExternal(domainID, workflowID) {
  metricTags = {"domain": domainName, "wftype": WFType}
  if enforced {
     scope.Tagged(metricTags).Counter("ratelimited_workflows").Inc()
     log.Info("rate limited by cache", wfID)
     return nil, Err429
  }
  scope.Tagged(metricTags).Counter("shadow_ratelimited_workflows").Inc()
  log.Info("rate limited but in shadow mode")
}

Reply:
Yeah, it's inside, so we dont have to have metric emission code everywhere